File: subprojects/client/src/main/groovy/com/canoo/dolphin/core/client/Dolphin.java
package com.canoo.dolphin.core.client;

import com.canoo.dolphin.core.client.comm.ClientConnector;

public class Dolphin {
    private static ClientModelStore clientModelStore;

    private static ClientConnector clientConnector;

    public static ClientConnector getClientConnector() {
        return clientConnector;
    }

    public static void setClientConnector(ClientConnector clientConnector) {
        Dolphin.clientConnector = clientConnector;
    }

    public static ClientModelStore getClientModelStore() {
        return clientModelStore;
    }

    public static void setClientModelStore(ClientModelStore clientModelStore) {
        Dolphin.clientModelStore = clientModelStore;
    }
}

File: dolphin-grails/web-app/src/Dolphin.js
define([
    'comm/ClientDolphin',
    'comm/ClientModelStore',
    'comm/HttpClientConnector'
], function (ClientDolphin, ClientModelStore, HttpClientConnector) {

    return function(serverUrl) {

        this.clientDolphin = new ClientDolphin();
        this.clientDolphin.setClientModelStore(new ClientModelStore(this.clientDolphin));

        var connector = new HttpClientConnector(this.clientDolphin, serverUrl);
        this.clientDolphin.setClientConnector(connector);

        this.getClientDolphin = function() {
            return this.clientDolphin;
        };

        console.log("dolphin started", serverUrl);

    };

});
